frontend/src/scenes/experiments/experimentLogic.tsx

```diff
@@ -1735,6 +1735,19 @@ export const experimentLogic = kea<experimentLogicType>([
                 return primaryMetricsLengthWithSharedMetrics > 0
             },
         ],
+        firstPrimaryMetric: [
+            (s) => [s.experiment],
+            (experiment: Experiment): ExperimentTrendsQuery | ExperimentFunnelsQuery | undefined => {
+                const primarySharedMetrics = experiment.saved_metrics.filter(
```

Comment:
I think this could be a little easier to read with something like

```
 if (experiment.metrics?.length) {
   return experiment.metrics[0];
 }

 const primaryMetric = experiment.saved_metrics
   ?.find(metric => metric.metadata.type === 'primary');

 if (primaryMetric) {
   return primaryMetric.query;
 }
 ```

Or it you want to go full weido
 ```
 experiment.metrics?.[0] ?? experiment.saved_metrics?.find(m => m.metadata.type === 'primary')?.query;
  ```